### PR TITLE
[R4ML-249] ml.preprocessor doesn't work with the use case of r4ml demo

### PR DESCRIPTION
**Pull request by aloknsingh**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: R4ML/R/r4ml.frame.R [hunk 1]
```diff
@@ -259,7 +259,8 @@ setMethod("r4ml.impute",
     constants <- list()
     constant_names <- list()
     
-    for(col in names(df_columns)){
+    num_cols <- df_columns[df_columns == "mean"]
```
Comment (bdwyer2) [inline R4ML/R/r4ml.frame.R:262]: What is the purpose of this? Why not just set `num_cols <- "mean"`?

File: R4ML/R/r4ml.frame.R [hunk 2]
```diff
@@ -259,7 +259,8 @@ setMethod("r4ml.impute",
...
+    for(col in names(num_cols)){
```
Comment (bdwyer2) [inline R4ML/R/r4ml.frame.R:263]: Why is a `for` loop needed?

File: R4ML/R/r4ml.frame.R (hunk 1 above)
Comment (aloknsingh) [reply R4ML/R/r4ml.frame.R:262]: This is the named list so we are collecting all of them which are the "mean" the output is not just the one element

File: R4ML/R/r4ml.frame.R (hunk 2 above)
Comment (aloknsingh) [reply R4ML/R/r4ml.frame.R:263]: see the previous comments.

**Pull request closed without merging**